system-tests/lib/performance-reporter.js

```diff
@@ -0,0 +1,64 @@
+const path = require('path')
+const chalk = require('chalk')
+const Libhoney = require('libhoney')
+
+const pkg = require('@packages/root')
+const ciProvider = require('@packages/server/lib/util/ci_provider')
+const { commitInfo } = require('@cypress/commit-info')
+
+class StatsdReporter {
+  constructor (runner) {
+    if (!process.env.HONEYCOMB_API_KEY) {
+      return
+    }
+
+    console.log(chalk.green('Reporting to honeycomb'))
+    this.honey = new Libhoney({
+      dataset: 'systemtest-performance',
+      writeKey: process.env.HONEYCOMB_API_KEY,
+    })
+
+    runner.on('test', (test) => {
+      test.wallclockStart = Date.now()
+    })
+
+    runner.on('test end', (test, done) => {
+      // Skipped tests never get a 'start' event, but they still get 'test end' somehow.
+      if (test.state === 'skipped') {
+        return
+      }
+
+      commitInfo().then((commitInformation) => {
+        const ciInformation = ciProvider.commitParams() || {}
+        const [, testTitle, browser] = test.title.match(/(.+?)(?: \[([a-z]+)\])?$/)
```

Comment:
What's this regex doing?

Reply:
Pulling the browser our of the test name. In system-tests.ts, there's the code
```
const testTitle = `${title} [${browser}]`
```
And I'm puling that back apart with this regex. Except some system tests aren't browser-specific - so the second block (" [firefox]") is optional. If absent, we get undefined for the browser and use the whole string as the testTitle.

Pulling it apart allows us to easily compare tests across browsers - having `e2e async timeouts / failing1` + `['electron', 'firefox']` is better than `e2e async timeouts / failing1 [electron]`, `e2e async timeouts / failing1 [firefox]`

I'll add a code comment explaining what's going on here.